src/feature/gui/gui-runner.c

```diff
@@ -506,21 +512,33 @@ void mGUIRun(struct mGUIRunner* runner, const char* path) {
 			if (guiKeys & (1 << mGUI_INPUT_SCREENSHOT)) {
 				mCoreTakeScreenshot(runner->core);
 			}
+			bool muteTogglePressed = guiKeys & (1 << mGUI_INPUT_MUTE_TOGGLE);
+			if (muteTogglePressed) {
+				mute = !runner->core->opts.mute;
```

Comment:
You should probably be either storing the old value of mute outside of the core or be pulling it directly from the runner config since the core's concept can get desynchronized, e.g. if mute is off but fast-forward mute is on, then you fast-forward and toggle mute it'll get confused. Since you're already polling that whenever it could change you can probably just simplify this as `mute = !mute`, but I'm not 100% sure about other contexts. I would recommend testing toggling mute both while fast-forwarding and while not fast-forwarding with fastForwardMute enabled and not enabled to make sure the correct behavior is always observed.

Putting the entire block of logic for how to set mute currently in one place after the frame limiter stuff will probably simplify the code for that a bit too.

Reply:
I already tested it before, but I did change this to `mute = !mute` just to be sure and tested it again. I tested all the scenarios you mentioned a couple of times and it worked as expected.